### DISCUSSION TITLE
Take a vertex from a set of connected vertices

### DISCUSSION THREAD

**Discussion by dinhphu2k1-gif** [JanusGraph/janusgraph#3448] [Q&A]
When i search vertex with a property, it return for me a set of connected vertices. And it takes quite a time, while i need onle one of them.
Note: maybe there are other vertices that are not connected and i need them too

![image](https://user-images.githubusercontent.com/81508954/210534938-8a6b4604-e232-4dfa-aa52-6e2995d2a34d.png)

**Comment by vtslab** [comment]
What is the query you used? How to recognize the vertex you want?

**Reply by dinhphu2k1-gif** [reply]
i use query "g.V().has(key, value)"

**Reply by dinhphu2k1-gif** [reply]
@vtslab 
problem it return many vertex connected with each other
and i don't want like this
if there are 2 connected graphs, then i want to return any 1 vertex in each graph then that's it

**Reply by dinhphu2k1-gif** [reply]
image above represent 1 connnected graph, and maybe there are many connected graphs like this

**Comment by vtslab** [comment]
In gremlin console, your query only returns a single vertex if the (key, value) pair is unique among the vertices. If many vertices have the same (key, value) pair, you have to add a second filter .has(key2, value2) to get the vertex you want.

Alternatively, the visualization tool that you use (which one?) might have some settings that result in additional queries.

**Comment by li-boxuan** [comment]
If I am understanding correctly, your problem is that `g.V().has(key, value)` returns two clusters of vertices, and you only want a random vertex from each cluster?

Try this:
Step 1. Extract a subgraph from your graph: https://tinkerpop.apache.org/docs/current/reference/#subgraph-step
Step 2. Find connected components (in your example, it would return two clusters) from the extracted subgraph: https://tinkerpop.apache.org/docs/current/recipes/#connected-components

Alternatively, you can first get all vertices by gremlin query, and then write your custom code to separate the two clusters

**Reply by dinhphu2k1-gif** [reply]
@li-boxuan i need real time so i think get all vertex is bad ideal

**Reply by li-boxuan** [reply]
I don’t think that’s ever possible without getting all vertices first.
